src/fastmcp/tools/tool_transform.py

```diff
@@ -798,3 +829,43 @@ def _function_has_kwargs(fn: Callable[..., Any]) -> bool:
         return any(
             p.kind == inspect.Parameter.VAR_KEYWORD for p in sig.parameters.values()
         )
+
+
+class ToolTransformConfig(FastMCPComponent):
```

Comment:
I'm not sure if this should inherit from FastMCPComponent - that's the base class for e.g. Tools Resources etc and consolidates logic like enabling / disabling. It could ultimately include fields that aren't good to have on the ToolTransformCOnfig (e.g. fields we dont expose for transforming). I think it's better to write out the transforable fields explicitly here.